src/python/correct_fastq.py

```diff
@@ -39,41 +39,44 @@ def get_barcodes(whitelist_file):
     return list(r1_barcodes), list(r2_barcodes), list(r3_barcodes)
 
 
-def create_barcode_dict(barcode_list):
+def create_barcode_dicts(barcode_list):
     """
-    Adds each barcode and its mismatch possibilities to the dictionary
+    Creates dictionaries containing exact match and 1-mismatch sequences
     """
-    barcode_dict = dict()
+    barcode_dict_exact  = dict() # [seq: seq]
+    barcode_dict_mismatch = dict() # [mismatch: seq]
     for barcode in barcode_list:
-        barcode_dict[barcode] = barcode  # exact match
+        barcode_dict_exact[barcode] = barcode # exact match
         for i, base in enumerate(barcode):
             for x in 'ACGTN':
                 if base != x:
                     # add mismatch possibilities at pos i
                     mismatch = barcode[:i] + x + barcode[i + 1:]
-                    barcode_dict[mismatch] = barcode
+                    barcode_dict_mismatch[mismatch] = barcode
+                    
+    return barcode_dict_exact, barcode_dict_mismatch
 
-    return barcode_dict
-
-def check_putative_barcode(barcode_str, barcode_dict, quality_str):
+def check_putative_barcode(barcode_str, barcode_dict_exact, barcode_dict_mismatch, quality_str):
```

Comment:
This function is used here and in raw_bam_to_fastq. That means that we need to always make sure that if we change one we will change also the other.  The best thing would be to put them in a util file and source it here.